src/panels/sell-panel/components/sell-pricechart/chart-data.ts

```diff
@@ -0,0 +1,78 @@
+import { formatTimeWithoutParenthesis } from "../../../../shared/utils/format-time";
+
+interface ChartData {
+  options: ApexCharts.ApexOptions;
+  series: ApexAxisChartSeries;
+}
+
+const getCurrentTimeInMinutes = () => {
+  const currentTime = new Date();
+  return currentTime.getHours() * 60 + currentTime.getMinutes();
+};
+
+const getPossibleNextYValue = (prevValue: number): number => {
+  const minValue = 5;
+  const maxValue = 250;
+  let possibleNextValue = prevValue - Math.ceil(Math.random() * 15) + Math.ceil(Math.random() * 15);
+  possibleNextValue = possibleNextValue < minValue ? minValue : possibleNextValue;
+  possibleNextValue = possibleNextValue > maxValue ? maxValue : possibleNextValue;
+  return possibleNextValue;
+};
+
+export const getXValues = (): string[] => {
+  const formattableTime = getCurrentTimeInMinutes();
+  const xAxisDateValues: string[] = [];
+  for (let i = 0; i < 10; i++) {
+    let prevTimeValue = formattableTime - i;
+    prevTimeValue = prevTimeValue < 0 ? prevTimeValue + 24 * 60 : prevTimeValue;
+    xAxisDateValues.unshift(formatTimeWithoutParenthesis(prevTimeValue));
+  }
+  return xAxisDateValues;
+};
+
+export const shiftXValues = (xValues: string[]): string[] => {
+  xValues.shift();
+  xValues.push(formatTimeWithoutParenthesis(getCurrentTimeInMinutes()));
```

Comment:
if browser tab sleeps it's possible that more time passes than a minute, which could lead to a bigger gap between values; consider generating the array on every update instead of shift (or check if this could be an issue)

Reply:
`getXValues` is used instead of `shiftXValues`; resolving comment